analysis/src/Commands.ml

```diff
@@ -178,100 +178,107 @@ let typeDefinition ~path ~pos ~debug =
     | Some location -> location |> Protocol.stringifyLocation)
 
 let references ~path ~pos ~debug =
-  let allLocs =
-    match Cmt.loadFullCmtFromPath ~path with
-    | None -> []
-    | Some full -> (
-      match References.getLocItem ~full ~pos ~debug with
-      | None -> []
-      | Some locItem ->
-        let allReferences = References.allReferencesForLocItem ~full locItem in
-        allReferences
-        |> List.fold_left
-             (fun acc {References.uri = uri2; locOpt} ->
-               let loc =
-                 match locOpt with
-                 | Some loc -> loc
-                 | None -> Uri.toTopLevelLoc uri2
-               in
-               Protocol.stringifyLocation
-                 {uri = Uri.toString uri2; range = Utils.cmtLocToRange loc}
-               :: acc)
-             [])
-  in
-  print_endline
-    (if allLocs = [] then Protocol.null
-     else "[\n" ^ (allLocs |> String.concat ",\n") ^ "\n]")
+  Eio_main.run (fun env ->
```

Comment:
Would a future goal be to have this `Eio_main` at the main entry point level of analysis tool?